src/Event/ImageProcessingListener.php

```diff
@@ -77,6 +77,12 @@ public function implementedEvents() {
 	 * @return boolean
 	 */
 	protected function _autoRotate($imageFile, $format) {
+		// Auto-rotate applies only to JPEG files (i.e. not to PNG, GIF, etc..)
+		$format = strtolower($format);
+		if($format != 'jpg' && $format != 'jpeg') {
```

Comment:
I added the space between if and (.

I'm not sure what you mean with !==. Isn't !== stronger comparison? I thought it's used for type-specific comparison. So, for instance, if you want to know if the variable is boolean or integer, you use === or !== otherwise you use == or != 

In this particular case, why do we need to do type-checking? $format will always be a string, no?

As for why JPEG only:

You are right. I was too lazy to check other compatible formats, but it turns out that the only other format that exif_read_data handle is TIFF. So, I added that.

http://php.net/manual/en/function.exif-read-data.php

We have to keep in mind that rotating and auto-rotating implies two different things. Yes, you can rotate any image files including PNG but this is not a situation where the user is manually editing an image, and telling us that he wants this particular image rotated for some reason.

Auto-rotate is for a situation where cameras, like iPhone's camera, do not actually rotate the image it captures but simply flag the orientation in the meta data (EXIF). So, in my opinion, auto-rotate should be turned on by default. Otherwise, there will be many photos uploaded and processed with the wrong orientation because this plugin will ignore orientation. If you try to upload any vertical/portrait photos taken with an iPhone, I think it will come in sideways (at least that is what is happening to me).

In other words, auto-rotation is for photos taken on cameras only. Because the default orientation of the camera sensors is horizontal, most cameras save all photos in horizontal orientation and simply note the orientation in the meta data.

If you are creating an image in Photoshop, or any image editing apps, this problem wouldn't exist. Since no camera (as far as I know) saves photos in PNG, there is no need for automatically correcting what the camera did.